src/test/java/com/aws/greengrass/lifecyclemanager/GreengrassServiceTest.java

```diff
@@ -116,4 +131,35 @@ void GIVEN_service_WHEN_dependencies_change_THEN_service_dependencies_updates()
         assertNull(aService.dependencies.get(dService));
         assertNotNull(aService.dependencies.get(eService));
     }
+
+    @Test
+    void GIVEN_service_WHEN_service_is_closed_THEN_service_dependency_watcher_is_removed() throws Exception {
+        // GIVEN service
+        Path configPath = Paths.get(this.getClass().getResource("services.yaml").toURI());
+        Configuration serviceConfiguration = new Configuration(context);
+        serviceConfiguration.read(configPath);
+        Topics root = serviceConfiguration.getRoot();
+        Topics serviceTopics = spy(root.findTopics(SERVICES_NAMESPACE_TOPIC, "B"));
+        Topic externalDependencyTopic = mock(Topic.class);
+
+        doReturn(context).when(serviceTopics).getContext();
+        doReturn(externalDependencyTopic).when(serviceTopics).createLeafChild(eq(SERVICE_DEPENDENCIES_NAMESPACE_TOPIC));
+        doReturn(externalDependencyTopic).when(externalDependencyTopic).dflt(any());
+        doReturn(Collections.emptyList()).when(externalDependencyTopic).getOnce();
+        lenient().doReturn(externalDependencyTopic).when(externalDependencyTopic).subscribe(any());
```

Comment:
Why lenient? if this is specifically being mocked for this test, then it either gets invoked or it doesn't

Reply:
I always run into a `Strict stubbing argument mismatch` if I do not use lenient

Reply:
Okay, I dont think this is required at all